core/src/main/java/org/elasticsearch/index/similarity/SimilarityService.java

```diff
@@ -35,11 +36,11 @@
 import java.util.function.BiFunction;
 
 public final class SimilarityService extends AbstractIndexComponent {
-
     public static final String DEFAULT_SIMILARITY = "BM25";
-    private final Similarity defaultSimilarity;
-    private final Similarity baseSimilarity;
-    private final Map<String, SimilarityProvider> similarities;
+
+    public static final Setting<Settings> SIMILARITY_SETTINGS = Setting.groupSetting("index.similarity.",
```

Comment:
I think what we should do here is to add functionality to specify something like this:

Instead of ` private static final Setting<Float> LAMBDA_SETTING = Setting.floatSetting("lambda", 0.1f, Setting.Property.Dynamic);` we would add ` private static final Setting<Float> LAMBDA_SETTING = Setting.floatSetting("index.similarity.*.lambda", 0.1f, Setting.Property.Dynamic);` and add the ability for making this setting _concrete_ via something like `Setting#concretSetting(name)`. For the updateConsumer API I guess we must add another method that accepts the generic setting and the concrete one to pass the checks in there? @jimferenczi what do you think?